compiler/options.nim

```diff
@@ -524,18 +525,19 @@ proc getOutFile*(conf: ConfigRef; filename: RelativeFile, ext: string): Absolute
   conf.outDir / changeFileExt(filename, ext)
 
 proc absOutFile*(conf: ConfigRef): AbsoluteFile =
+  if false:
+    doAssert not conf.outDir.isEmpty
+    doAssert not conf.outFile.isEmpty
+    # xxx: fix this pre-existing bug causing `SuccessX` error messages to lie
+    # for `jsonscript`
   result = conf.outDir / conf.outFile
   when defined(posix):
-    if dirExists(result.string):
-      result.string.add ".out"
+    if dirExists(result.string): result.string.add ".out"
 
 proc prepareToWriteOutput*(conf: ConfigRef): AbsoluteFile =
   ## Create the output directory and returns a full path to the output file
-  createDir conf.outDir
-  result = conf.outDir / conf.outFile
-  when defined(posix):
-    if dirExists(result.string):
-      result.string.add ".out"
```

Comment:
Hey, wait a second! That was good code! ``nim c -r foo`` where foo is also a directory must continue to work!

Reply:
Never mind, I misread.